storybook/story-components/doc-item.js

```diff
@@ -1,76 +1,78 @@
 import React from 'react';
 
 import DocText from './doc-text';
+import DocCode from './doc-code';
+import InlineCode from './inline-code';
 import insertBetween from './insert-between';
 import { StyleSheet, Text, View } from 'react-native';
+import { size } from './platform-styles';
 
 const Divider = () => <View style={styles.verticalDivider} />;
 
 const createDescription = description => {
   const nodeList = React.Children.toArray(description);
-  let content;
-  if (nodeList.length === 1) {
-    content = <Text>{nodeList}</Text>;
-  } else {
-    content = insertBetween(() => <Divider key={Math.random()} />, nodeList);
-  }
-  return <Text style={styles.text}>{content}</Text>;
+  return typeof description === 'string' ? (
+    <Text>{description}</Text>
+  ) : (
+    insertBetween(() => <Divider key={Math.random()} />, nodeList)
+  );
 };
 
-const DocItem = ({ description, example = {}, name, typeInfo, label }) => (
+const DocItem = ({ description, example = {}, name, typeInfo, label, required, defaultValue }) => (
   <View style={styles.example}>
-    {name && (
-      <DocText style={styles.title}>
-        <PropText label={label} name={name} typeInfo={typeInfo} />
-      </DocText>
+    {!!name && (
```

Comment:
There's likely a linting rule we could have in place to avoid `!!`. It's not intuitive for everyone and requires extra cognitive load as opposed to seeing something more verbose that spells out what's happening.

Reply:
Hmm, I thought this was pretty standard these days. I don't see how linting will help here as it is to prevent a react native issue - It has to resolve to true/false and not truthy/falsy. I could use `Boolean(name)` instead?

Reply:
This isn't a `React Native` specific thing though. This is more so how the logic is being represented using what some might refer to as a "Trick".

Reply:
How would you prefer this be handled? The style I'm using keeps the code declarative... although, yes, it requires understanding that `!!` turns truthy into true and that `true && 'Some text' === 'Some text'`. I wouldn't say they are tricks, but that they are advanced uses of operators and not really any different from using `+` to perform addition - in that once you know how the operator works it is easy to read and understand. Neither depends on javascript weirdness

I could use a helper: (untested)

```
const renderIfPopulated = (content, component) => 
  (content !== undefined 
    && content !== null 
    && (typeof content !== 'string' || content.length > 0) 
      ? component : null

...

    {renderIfPopulated(name, 
      <PropText label={label} name={name} typeInfo={typeInfo} required={required} defaultValue={defaultValue} />
    )}

``` 
I think this is harder to read and comprehend

Reply:
It shouldn't be the case that it could be `null` || `undefined`. It's likely only that `{name && ()}` or `name !== undefined && ()` would suffice but is instantly more readable to someone new to the code.

Yes, `!!` is fine if it's something you use regularly. But we normally avoid using it for readability/reduced cognitive load when reviewing/reading code.